src/Avalonia.Controls/ToolTip.cs

```diff
@@ -106,24 +106,28 @@ private static void ShowToolTip(Control control)
             if (control != null && control.IsVisible && control.GetVisualRoot() != null)
             {
                 var cp = (control.GetVisualRoot() as IInputRoot)?.MouseDevice?.GetPosition(control);
-                var position = control.PointToScreen(cp ?? new Point(0, 0)) + new Vector(0, 22);
 
-                if (s_popup == null)
+                if (cp.HasValue && control.IsVisible && new Rect(control.Bounds.Size).Contains(cp.Value))
```

Comment:
The issue is whether we should show the tooltip if the mouse is captured by the control _with_ the tooltip and the mouse is moved away from the control. Not sure what the correct behavior here is tbh, i'd need to check with other frameworks.